presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkNativeExecutionShuffleManager.java

```diff
@@ -71,19 +73,27 @@ private static <T> T instantiateClass(String className, SparkConf conf)
         }
     }
 
+    protected void registerShuffleHandle(BaseShuffleHandle handle, int mapId)
+    {
+        partitionIdToShuffleHandle.put(mapId, handle);
+        shuffleIdToBaseShuffleHandle.put(handle.shuffleId(), handle);
+    }
+
     @Override
     public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, int numMaps, ShuffleDependency<K, V, C> dependency)
     {
-        shuffleNumMaps.put(shuffleId, numMaps);
         return fallbackShuffleManager.registerShuffle(shuffleId, numMaps, dependency);
     }
 
     @Override
     public <K, V> ShuffleWriter<K, V> getWriter(ShuffleHandle handle, int mapId, TaskContext context)
     {
-        shuffleDependencyMap.put(mapId, handle);
-        int shuffleId = handle.shuffleId();
-        return new EmptyShuffleWriter<>(getNumOfPartitions(shuffleId));
+        if (!(handle instanceof BaseShuffleHandle)) {
+            throw new RuntimeException("Only support ShuffleHandle inheriting from BaseShuffleHandle");
+        }
+        BaseShuffleHandle<?, ?, ?> baseShuffleHandle = (BaseShuffleHandle<?, ?, ?>) handle;
+        registerShuffleHandle(baseShuffleHandle, mapId);
+        return new EmptyShuffleWriter<>(baseShuffleHandle.dependency().partitioner().numPartitions());
```

Comment:
Is there a TODO here ? If so, can you please add a TODO.

Reply:
I'm not exactly sure. But based on my understanding the EmptyShuffleWriter here is a placeholder to satisfy the Java framework. The real shuffle writer is in C++.